R/transform_log.R

```diff
@@ -0,0 +1,18 @@
+#' transform_log 
+#' Transform numerical values into their log values
+#' @param x A vector
+#' @return logarithm of x
+#' @examples
+#' transform_log(c(NA,0,-1,exp(2)))
+#'@export 
+transform_log<-function(x){
+  if(!is.numeric(x))stop("function is expecting only numeric values")
+  x_nan<-is.na(x)
+  x[x_nan]<-1
+  ifelse(x<0,"OK", warning("input vector contains negative values, turned into NA"))
```

Comment:
This statement gives a warning for **EVERY** negative value in `x`.
It would also be sufficient to do a 
```
if(any(x < 0)) {
  warning("input vector contains negative values, turned into NA")
}
```